src/Lecture3.hs

```diff
@@ -213,8 +252,24 @@ types that can have such an instance.
 -- instance Foldable Weekday where
 -- instance Foldable Gold where
 -- instance Foldable Reward where
--- instance Foldable List1 where
--- instance Foldable Treasure where
+instance Foldable List1 where
+  foldr :: (a -> b -> b) -> b -> List1 a -> b
+  foldr f z (List1 a []) = f a z
+  foldr f z (List1 a (x : xs)) = f a (foldr f z (x : xs))
```

Comment:
You don't need to pattern match on the `x : xs` list here and can reuse the `foldr` instance for ordinary list immediately 🙂